workers/unity/Packages/com.improbable.gdk.tools/Common.cs

```diff
@@ -67,52 +81,37 @@ public static string GetThisPackagePath()
             }
         }
 
-        public static int RunProcess(string command, params string[] arguments)
+        private static string DiscoverSpatialLocation()
         {
-            var info = new ProcessStartInfo(command, string.Join(" ", arguments))
-            {
-                CreateNoWindow = true,
-                RedirectStandardError = true,
-                RedirectStandardOutput = true,
-                UseShellExecute = false,
-                WorkingDirectory = Path.GetFullPath(Path.Combine(Application.dataPath, ".."))
-            };
-
-            using (var process = Process.Start(info))
+            var pathValue = Environment.GetEnvironmentVariable("PATH");
+            if (pathValue == null)
             {
-                if (process == null)
-                {
-                    throw new Exception(
-                        $"Failed to run {info.FileName} {info.Arguments}\nIs the .NET Core SDK installed?");
-                }
-
-                process.EnableRaisingEvents = true;
-
-                process.OutputDataReceived += OnReceived;
-                process.ErrorDataReceived += OnErrorReceived;
-
-                process.BeginOutputReadLine();
-                process.BeginErrorReadLine();
+                return string.Empty;
+            }
 
-                process.WaitForExit();
-                return process.ExitCode;
+            var fileName = "spatial";
+            if (Application.platform == RuntimePlatform.WindowsEditor)
+            {
+                fileName = Path.ChangeExtension(fileName, ".exe");
             }
-        }
 
-        private static void OnReceived(object sender, DataReceivedEventArgs args)
-        {
-            if (!string.IsNullOrEmpty(args.Data))
+            var splitPath = pathValue.Split(Path.PathSeparator);
+
+            if (Application.platform == RuntimePlatform.OSXEditor && !splitPath.Contains(UsrLocalBinDir))
             {
-                Debug.Log(args.Data);
+                splitPath = splitPath.Union(new[] { UsrLocalBinDir }).ToArray();
             }
-        }
 
-        private static void OnErrorReceived(object sender, DataReceivedEventArgs args)
-        {
-            if (!string.IsNullOrEmpty(args.Data))
+            foreach (var path in splitPath)
             {
-                Debug.LogError(args.Data);
+                var testPath = Path.Combine(path, fileName);
+                if (File.Exists(testPath))
+                {
+                    return testPath;
+                }
             }
+
+            return string.Empty;
```

Comment:
`return splitPaths.Select(path => Path.Combine(path, fileName)).FirstOrDefault(testPath => File.Exists(testPath)) ?? String.Empty;` 😉 

`FirstOrDefault` will return null if its not there - so `?? String.Empty` ensures we return a valid string.